src/o365/graph/commands/teams/teams-add.ts

```diff
@@ -0,0 +1,197 @@
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId?: string;
+  name?: string;
+  description?: string;
+}
+
+class GraphTeamsAddCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Add a new Microsoft Teams team.';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        return args.options.groupId ? this.CreateGroupTeamRequest(cmd,args) : 
+                                      this.CreateTeamRequest(cmd,args);
+      })
+      .then((res: any): void => {
+        // get the teams id from the response header.
+        var teamsId = this.GetTeamsIdFromResponse(res, args);
+        cmd.log({Team:teamsId});
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+        cb();
+      }, (err: any): void =>{ 
+        this.handleRejectedODataJsonPromise(err, cmd, cb)
+      });
+  }
+
+  private GetTeamsIdFromResponse(res:any, args: CommandArgs) : string {
+    let teamsRspHdrRegEx;
+    if(!args.options.groupId) {
+      teamsRspHdrRegEx = /(teams\(')([0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12})('\))/i.exec(res.headers.location);
+    }
+    else {
+      teamsRspHdrRegEx = /(team\(')([0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12})('\))/i.exec(res.headers.location);
+    }
+
+    if(teamsRspHdrRegEx != null && teamsRspHdrRegEx.length == 4)
+    {
+      return teamsRspHdrRegEx[2];
+    }
+
+    return '';
+  }
+
+  private CreateTeamRequest(cmd : CommandInstance, args: CommandArgs) : request.RequestPromise {
+    const teamsEndpoint: string = `${auth.service.resource}/beta/teams`;
+    const teamsRequestBody = {
+      'template@odata.bind': 'https://graph.microsoft.com/beta/teamsTemplates/standard',
+      displayName: args.options.name,
+      description: args.options.description
+    };
+
+
+    const requestOptions: any = {
+      url: teamsEndpoint,
+      resolveWithFullResponse: true ,
+      headers: Utils.getRequestHeaders({
+        authorization: `Bearer ${auth.service.accessToken}`,
+        accept: 'application/json;odata.metadata=none',
+        'content-type': 'application/json;odata=nometadata'
+      }),
+      body: teamsRequestBody,
+      json: true
+    };
+
+    if (this.debug) {
+      cmd.log('Executing web request...');
+      cmd.log(requestOptions);
+      cmd.log('');
+    }
+
+    return request.post(requestOptions);
+  }
+
+  private CreateGroupTeamRequest(cmd : CommandInstance, args: CommandArgs) : request.RequestPromise {
```

Comment:
Since this method already executes the request, we should rename it to `CreateGroupTeam`